### PR TITLE
Add DNS A record named autoip

### PR DESCRIPTION
**Pull request by mikev**
Add DNS A record named autoip, which is derived from the containerip or agentip, depending on the existence of port mappings.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,4 +16,5 @@ _deps
 _plugins
```
Comment (sargun) [inline .gitignore:16]: Mind breaking this into a separate commit?

File: apps/navstar_dns/priv/axfr_records [hunk 2]
```diff
@@ -1,41 +1,41 @@
...
-            {dns_rr, <<"root.ns1.mesos.thisdcos.directory">>, 1, 1, 5,
-                {dns_rrdata_a, {198, 51, 100, 2}}},
-            {dns_rr, <<"root.ns1.mesos.thisdcos.directory">>, 1, 1, 5,
-                {dns_rrdata_a, {198, 51, 100, 3}}},
-            {dns_rr, <<"slave.mesos.thisdcos.directory">>, 1, 1, 5,
-                {dns_rrdata_a, {10, 0, 3, 101}}},
-            {dns_rr, <<"slave.mesos.thisdcos.directory">>, 1, 1, 5,
-                {dns_rrdata_a, {10, 0, 5, 155}}}].
\ No newline at end of file
+[{dns_rr,<<"_framework._tcp.marathon.mesos.thisdcos.directory">>,1,
```
Comment (sargun) [inline apps/navstar_dns/priv/axfr_records]: Thanks for this formatting. Mind breaking the reformatting of these files into their own commit?

File: apps/navstar_dns/src/navstar_dns_poll_fsm.erl [hunk 3]
```diff
@@ -694,6 +729,69 @@ zone_records_fake_test() ->
...
+    {<<"agentip">>, {1, 2, 3, 11}},
+    {<<"agentip">>, {1, 2, 3, 11}},
+    {<<"agentip">>, {1, 2, 3, 11}},
+    {<<"agentip">>, {1, 2, 3, 12}},
+    {<<"agentip">>, {1, 2, 3, 11}},
+    {<<"agentip">>, {1, 2, 3, 11}},
+    {<<"agentip">>, {1, 2, 3, 11}},
+    {<<"agentip">>, {1, 2, 3, 11}}].
+
+zone_records_autoip_test() ->
```
Comment (sargun) [inline apps/navstar_dns/src/navstar_dns_poll_fsm.erl:775]: These tests are great. Does it make sense to get a state.json file that would trigger the heuristic? I think all it's a matter of doing is launching an application on Docker bridge, docker host, and docker bridge, and adding a port mapping. You should be able to do this with any DC/OS.

File: apps/navstar_dns/src/navstar_dns_poll_fsm.erl [hunk 4]
```diff
@@ -401,17 +400,52 @@ mesos_master_poll() ->
...
+-spec(task_ip_by_network_infos(task(), binary()) -> {binary(), inet:ip4_address()}).
+task_ip_by_network_infos(
+    #task{statuses = [#task_status{container_status = #container_status{network_infos = NetworkInfos}}|_]}, Label) ->
+    [#network_info{ip_addresses = IPAddresses}|_] = NetworkInfos,
+    [#ip_address{ip_address = IP}|_] = IPAddresses,
+    {Label, IP};
+task_ip_by_network_infos(Task, Label) ->
+   task_ip_by_agent(Task, Label).
+
+-spec(task_ip_autoip(task()) -> {binary(), inet:ip4_address()}).
```
Comment (sargun) [inline apps/navstar_dns/src/navstar_dns_poll_fsm.erl:441]: lgtm

File: apps/navstar_dns/priv/state3_records [hunk 5]
```diff
@@ -1,71 +1,102 @@
...
-                1, 1, 5,
-                {dns_rrdata_a, {10, 10, 0, 109}}},
-            {dns_rr, <<"integration-test-proxy-to-minuteman-vip-597177d7b90b4cfaac21ccf.marathon.containerip.dcos.thisdcos.directory">>,
-                1, 1, 5,
-                {dns_rrdata_a, {10, 10, 0, 109}}},
-            {dns_rr, <<"integration-test-proxy-to-minuteman-vip-597177d7b90b4cfaac21ccf.marathon.slave.dcos.thisdcos.directory">>,
-                1, 1, 5,
-                {dns_rrdata_a, {10, 10, 0, 109}}}].
\ No newline at end of file
+[{dns_rr,<<"_integration-test-app-with-minuteman-vip-597177d7b90b4cfaac21ccf._tcp.marathon.agentip.dcos.thisdcos.directory">>,
```
Comment (sargun) [inline apps/navstar_dns/priv/state3_records]: If it's not too difficult, can you put spaces after your commas?

**Comment by sargun** [comment]
When you break up the commits I'd prefer to have them in the order:
- Add .gitignore
- Reformat files
- Add new AutoIP entires to tests, and the relevant code

I can understand if splitting 2 and 3 is hard, so if you can't do that let me know.

File: apps/navstar_dns/src/navstar_dns_poll_fsm.erl [hunk 6]
```diff
@@ -401,17 +400,52 @@ mesos_master_poll() ->
...
+task_ip_autoip(Task = #task{container = #container{type = docker, docker = #docker{port_mappings = []}}}) ->
```
Comment (sargun) [inline apps/navstar_dns/src/navstar_dns_poll_fsm.erl:442]: Do you know how this works for unified containerizer? Maybe @jieyu can comment?

File: .gitignore (hunk 1 above)
Comment (mikev) [inline .gitignore:16]: Done!

File: apps/navstar_dns/src/navstar_dns_poll_fsm.erl (hunk 6 above)
Comment (mikev) [inline apps/navstar_dns/src/navstar_dns_poll_fsm.erl:442]: It is a good question...  We're going to test this scenario on Monday.

File: apps/navstar_dns/priv/state3_records (hunk 5 above)
Comment (mikev) [inline apps/navstar_dns/priv/state3_records]: Done!

File: apps/navstar_dns/priv/axfr_records (hunk 2 above)
Comment (mikev) [inline apps/navstar_dns/priv/axfr_records]: Done.

File: apps/navstar_dns/src/navstar_dns_poll_fsm.erl (hunk 6 above)
Comment (sargun) [inline apps/navstar_dns/src/navstar_dns_poll_fsm.erl:442]: Do you want me to wait to merge before that? And you can submit a follow on patch, or good to merge now?

**Comment by sargun** [comment]
Looks good to me @mikev. Just checking if I'm good to merge. If so, do you mind squashing the commit down to one commit? Or I let Github do that otherwise.

**Pull request merged**